File: appAndroid/common-compose/src/main/java/com/gchristov/newsfeed/commoncompose/theme/ContentColor.kt
package com.gchristov.newsfeed.commoncompose.theme

import androidx.compose.runtime.Composable
import androidx.compose.runtime.Immutable
import androidx.compose.runtime.staticCompositionLocalOf
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.colorResource
import com.gchristov.newsfeed.commoncompose.R

@Immutable
data class ContentColors(
    val action: Color,
    val primary: Color,
    val secondary: Color,
    val warning: Color,
)

internal val LocalContentColors = staticCompositionLocalOf {
    ContentColors(
        action = Color.Unspecified,
        primary = Color.Unspecified,
        secondary = Color.Unspecified,
        warning = Color.Unspecified,
    )
}

@Composable
internal fun lightContentColors() = ContentColors(
    action = colorResource(R.color.red),
    primary = colorResource(R.color.black),
    secondary = colorResource(R.color.black_3),
    warning = colorResource(R.color.yellow),
)

@Composable
internal fun darkContentColors() = ContentColors(
    action = colorResource(R.color.red),
    primary = colorResource(R.color.white),
    secondary = colorResource(R.color.gray_3),
    warning = colorResource(R.color.yellow),
)

File: ios/CommonSwiftUi/Sources/Theme/ContentColor.swift
import SwiftUI
import CommonDesign

public struct ContentColors {
    public let action: Color
    public let primary: Color
    public let secondary: Color
    public let warning: Color
}

func lightContentColors() -> ContentColors {
    return ContentColors(
        action: Color.appRed,
        primary: Color.appBlack,
        secondary: Color.appBlack3,
        warning: Color.appYellow
    )
}

func darkContentColors() -> ContentColors {
    return ContentColors(
        action: Color.appRed,
        primary: Color.appWhite,
        secondary: Color.appGray3,
        warning: Color.appYellow
    )
}
